Sources/Articles/Documentation.docc/2025/jwt-kit/jwt-kit.md

```diff
@@ -0,0 +1,198 @@
+
+Other than HMAC, JWTKit also supports **ECDSA**, **EdDSA** and **RSA** keys. These are asymmetric signing algorithms, which means the key used to sign the JWT is different from the key used to verify it. This article won't go into which algorithm is best as each one has its own use case, however it's important to note that RSA is the least recommended of the listed ones due to different reasons. That's why JWTKit hides RSA keys behind an `Insecure` namespace.
+
+## Creating a JWT
+
+Once you have a ``JWTKeyCollection`` object, you can use it to "create" a JWT. Creating a JWT means signing the payload with a key, in this case one from the collection. The payload is the data we want to transmit securely:
+
+<!-- @Snippet(path: "site/Snippets/JWTKit", slice: payload_struct) -->
+
+```swift
+struct TestPayload: JWTPayload {
+    var expiration: ExpirationClaim
+    var issuer: IssuerClaim
+
+    enum CodingKeys: String, CodingKey {
+        case expiration = "exp"
+        case issuer = "iss"
+    }
+
+    func verify(using key: some JWTAlgorithm) throws {
+        try self.expiration.verifyNotExpired()
+    }
+}
+```
+
+In this example, we define a ``TestPayload`` struct that conforms to the ``JWTPayload`` protocol. This protocol requires us to implement the `verify(using:)` method, which includes optional additional validation logic that can be performed when creating the JWT. In this case, we're verifying that the token has not expired.
+The properties of the struct are the claims we want to include in the JWT. JWTKit provides a number of built-in claims, such as ``ExpirationClaim`` and ``IssuerClaim``, which are commonly used in JWTs. JWTKit supports the [seven registered claims](https://datatracker.ietf.org/doc/html/rfc7519#section-4.1) defined in the JWT specification, but you can also define custom claims if needed. 
+
+> Note: in the example, `CodingKeys` are defined to map Swift property names to the JSON keys used in the JWT. This means that in the JWT the expiration claim will be named `exp` and the issuer claim will be named `iss`, as per the JWT specification.
+
+To create a JWT with this payload, you can create a new instance of the payload and use the key collection to sign it:
+
+<!-- @Snippet(path: "site/Snippets/JWTKit", slice: jwt_sign) -->
+
+```swift
+let payload = TestPayload(
+    expiration: .init(value: .init(timeIntervalSinceNow: 3600)),
+    issuer: "myapp.com"
+)
+
+let token = try await keyCollection.sign(payload)
+```
+
+This will create a token that looks like the one we showed earlier. The token is now ready to be transmitted to the other party.
+
+## Validating a JWT
+
+Once you receive a JWT, you can use the key collection to validate it. Validation involves verifying the signature of the token and checking the claims in the payload:
```

Comment:
```suggestion
Once you receive a JWT, you can use the key collection to verify it. Verifying a JWT involves verifying the signature of the token and then checking the claims in the payload:
```

Should we rename **validate** to **verify** to stay consistent with your AI surface?